cdap-data-fabric/src/main/java/co/cask/cdap/data/format/AvroRecordFormat.java

```diff
@@ -37,12 +39,13 @@
   private org.apache.avro.Schema avroSchema;
   private DecoderFactory decoderFactory = DecoderFactory.get();
   private DatumReader<GenericRecord> datumReader;
+  private BinaryDecoder binaryDecoder;
 
   @Override
   public GenericRecord read(ByteBuffer input) {
     try {
       // TODO: CDAP-1199 look at reusing decoder and generic record
-      return datumReader.read(null, decoderFactory.binaryDecoder(Bytes.toBytes(input), null));
+      return datumReader.read(null, decoderFactory.binaryDecoder(Bytes.toBytes(input), binaryDecoder));
```

Comment:
that would be a bigger change. how about I open another jira and we do that separately?